src/analysis/normed_space/affine_isometry.lean

```diff
@@ -586,3 +588,42 @@ begin
   rw this,
   simp only [homeomorph.comp_is_open_map_iff, homeomorph.comp_is_open_map_iff'],
 end
+
+local attribute [instance, nolint fails_quickly] affine_subspace.to_normed_add_torsor
+local attribute [instance, nolint fails_quickly] affine_subspace.nonempty_map
+local attribute [instance, nolint fails_quickly] affine_subspace.to_add_torsor
+
+include V₁
+omit V
+
+/-- Restricts an affine isometry to an affine isometry equivalence between a nonempty affine
+subspace `E` and its image. -/
+noncomputable def affine_isometry.restrict_to_equiv
+  (E : affine_subspace 𝕜 P₁) [nonempty E]
+  (φ : P₁ →ᵃⁱ[𝕜] P₂) : E ≃ᵃⁱ[𝕜] E.map φ.to_affine_map :=
```

Comment:
Do we have the linear isometry version of this statement anywhere? We have the regular linear version as `submodule.equiv_map_of_injective`. I think it is worth having both, since then you can define this map in terms of the other one.

Reply:
And I don't think that a linear isometry version already exists.

Reply:
I think I'm asking the wrong question, `affine_isometry` isn't define the way I thought it was.

What I meant to suggest was first defining
```lean
/--
The affine version of `submodule.equiv_map_of_injective `.

TODO: also define a computable `affine_equiv.affine_subspace_map` version -/
noncomputable def affine_subspace.equiv_map_of_injective
  (E : affine_subspace 𝕜 P₁) [nonempty E]
  (φ : P₁ →ᵃ[𝕜] P₂) (hφ : function.injective φ) : E ≃ᵃ[𝕜] E.map φ :=
{ linear :=
    (E.direction.equiv_map_of_injective φ.linear (φ.injective_iff_linear_injective.mpr hφ)).trans
      (linear_equiv.of_eq _ _ (affine_subspace.map_direction _ _).symm),
  map_vadd' := λ p v, subtype.ext $ φ.map_vadd p v,
  .. equiv.set.image _ (E : set P₁) hφ }
  ```

Reply:
I see. I added this function now, choosing a slightly shorter proof that relies on `affine_equiv.of_bijective`, hoping this is also fine.

Reply:
The nice thing about the spelling above is that `@[simps]` generates a reasonable looking lemma for `.linear`; so I think I'd prefer my slightly longer proof.